modules/objdetect/src/hog.cpp

```diff
@@ -1453,6 +1453,7 @@ void HOGDescriptor::detect(const Mat& img,
     Size winStride, Size padding, const std::vector<Point>& locations) const
 {
     hits.clear();
+    weights.clear();
```

Comment:
O I followed up on this fix in 2.4, where it was only 1 line, but like remarked in the PR history, in master there are several `hits.clear()` lines that should be followed by the `weights.clear()` line.